### PR TITLE
portworx scenario

### PR DESCRIPTION
**Pull request by mumutyal**

### CODE REVIEW

File: block-storage-attacher/tests/e2e/e2e-tests/block_volume_attach.go [hunk 1]
```diff
@@ -144,6 +155,69 @@ var _ = framework.KubeDescribe("[Feature:Block_Volume_Attach_E2E]", func() {
...
+				Expect(err).NotTo(HaveOccurred())
+			}
+			framework.KubeDescribe("Test mount point: Create mount point, read, write...", func() {
+				//It("PRESTAGE: Test mount point: Create mount point, read, write... [Serial]", func() {
+				By("Creating a claim with a dynamic provisioning annotation")
+				claim := commontest.NewClaim(ns, "portworx-sc", "2Gi")
+				defer func() {
+					c.Core().PersistentVolumeClaims(ns).Delete(claim.Name, nil)
+				}()
+				claim, err := c.Core().PersistentVolumeClaims(ns).Create(claim)
```
Comment (akgunjal) [inline block-storage-attacher/tests/e2e/e2e-tests/block_volume_attach.go]: The PVC will take some time to create like around 2mins if its success case. You may need to keep checking status and then timeout before moving ahead else this may fail every time even if its successful.

File: block-storage-attacher/tests/e2e/e2e-tests/portworx_pvc.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+kind: PersistentVolumeClaim
```
Comment (akgunjal) [inline block-storage-attacher/tests/e2e/e2e-tests/portworx_pvc.yaml]: Is this PVC used anywhere? I didnt find any place in e2e where its used.

File: block-storage-attacher/tests/e2e/e2e-tests/block_volume_attach.go [hunk 3]
```diff
@@ -144,6 +155,69 @@ var _ = framework.KubeDescribe("[Feature:Block_Volume_Attach_E2E]", func() {
...
+			}
+			Expect(err).NotTo(HaveOccurred())
+
+			/* PVC Creation */
+
+			By("PVC  Creation")
+			if filestatus == true {
+				filepatharg := fmt.Sprintf("%s", portworxpvcpath)
+				filepatharg2 := fmt.Sprintf("%s", portworxscpath)
+				cmd := exec.Command(pvscriptpath, filepatharg, "portworxpvcreate", filepatharg2)
```
Comment (akgunjal) [inline block-storage-attacher/tests/e2e/e2e-tests/block_volume_attach.go]: I dont see `filepatharg` is being used. Is this param needed to be passed?

File: block-storage-attacher/tests/e2e/e2e-tests/portworx_values.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+kind: PersistentVolumeClaim
```
Comment (akgunjal) [inline block-storage-attacher/tests/e2e/e2e-tests/portworx_values.yaml]: Is this PVC used anywhere? I didnt find any place in e2e where its used.

File: block-storage-attacher/tests/e2e/e2e-tests/utilscript.sh [hunk 5]
```diff
@@ -25,6 +37,17 @@ then
     export NOD_IP=$3
     bx sl block access-revoke -p $NOD_IP $VOL_ID
     bx sl  block volume-cancel -f --immediate $VOL_ID
+elif [ "$2" = "portworxpvcreate" ]
+then
+    install_portworx_plugin
+    check_portworx_pod_state "portworx"
+    echo "BlockVolumeAttacher-Volume-Test: PortWorx Plugin-Installation: PASS" >> $E2E_PATH/e2eTests.txt
+    export CLSFILE=$3
+    kubectl  create -f $CLSFILE
```
Comment (akgunjal) [inline block-storage-attacher/tests/e2e/e2e-tests/utilscript.sh]: You can use `$3` here directly instead of doing `export`.

File: block-storage-attacher/tests/e2e/scripts/common.sh [hunk 6]
```diff
@@ -491,3 +513,41 @@ function install_blockvolume_plugin {
...
+    echo "Installing helm chart portworx plugin  .."
+	# INSTALL HELM TILLER (Attempt again, if already installed)
+	echo "Initialize tiller AND Wait till running"
+	helm init --force-upgrade
+	check_pod_state "tiller-deploy"
+	
+        # INSTALL/UPGRADE HELM CHART
+	helm_values_override="--set etcdEndPoint=$ETCD_SET1;$ETCD_SET2,clusterName=$(uuidgen),etcd.credentials=$ETCDCREDS,usedrivesAndPartitions=true,usefileSystemDrive=true"
+	#helm_install_cmd="helm install $helm_values_override $PORTWROX_HELM_CHART --name portworx"
+	helm_install_cmd="helm install  $PORTWROX_HELM_CHART --name portworx"
```
Comment (akgunjal) [inline block-storage-attacher/tests/e2e/scripts/common.sh]: (1) The helm chart path `$PORTWROX_HELM_CHART` is having a typo. It should be set to `$PORTWORX_HELM_CHART`.

(2) The `helm_values_override` is not used. You need to use them to set values.yaml with updated values.

File: block-storage-attacher/tests/e2e/scripts/common.sh [hunk 7]
```diff
@@ -491,3 +513,41 @@ function install_blockvolume_plugin {
...
+        exit 1
+    fi
+    echo "Installing helm chart portworx plugin  .."
+	# INSTALL HELM TILLER (Attempt again, if already installed)
+	echo "Initialize tiller AND Wait till running"
+	helm init --force-upgrade
+	check_pod_state "tiller-deploy"
+	
+        # INSTALL/UPGRADE HELM CHART
+	helm_values_override="--set etcdEndPoint=$ETCD_SET1;$ETCD_SET2,clusterName=$(uuidgen),etcd.credentials=$ETCDCREDS,usedrivesAndPartitions=true,usefileSystemDrive=true"
```
Comment (akgunjal) [inline block-storage-attacher/tests/e2e/scripts/common.sh]: You need to also set the `drives` parameter of values.yaml with device path (example `/dev/dm-1`) found in the `kubectl describe pv` output of block attacher PV object. You can see it in annotations and fetch from there.

File: block-storage-attacher/tests/e2e/conf/prod-us-east.conf [hunk 8]
```diff
@@ -54,6 +54,8 @@ KUBECONFIG=$KUBECONFIG			# Read from environment(incase of dev executions)
 HELM_CHART_REPO=$HELM_CHART_REPO	
 HELM_CHART=$GOPATH/src/github.com/IBM/ibmcloud-storage-utilities/block-storage-attacher/helm/ibm-block-storage-attacher/
 
+PORTWORX_HELM_CHART=$GOPATH/src/github.com/IBM/ibmcloud-storage-utilities/portworx_helm/helm/charts/portworx
```
[the remark this answers is not in the document]
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/conf/prod-us-east.conf]: DOne

File: block-storage-attacher/tests/e2e/e2e-tests/block_volume_attach.go (hunk 1 above)
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/e2e-tests/block_volume_attach.go]: The call returns only after fail/complete success because it will return the response body with PVC details

File: block-storage-attacher/tests/e2e/e2e-tests/portworx_pvc.yaml (hunk 2 above)
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/e2e-tests/portworx_pvc.yaml]: Earlier I was creating the PVC using the file . Later changed the logic I will remove it

Comment (mumutyal) [reply block-storage-attacher/tests/e2e/e2e-tests/portworx_pvc.yaml]: Done

File: block-storage-attacher/tests/e2e/e2e-tests/block_volume_attach.go (hunk 3 above)
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/e2e-tests/block_volume_attach.go]: Removed

File: block-storage-attacher/tests/e2e/e2e-tests/utilscript.sh (hunk 5 above)
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/e2e-tests/utilscript.sh]: Just for readbility in the script used

File: block-storage-attacher/tests/e2e/scripts/common.sh (hunk 6 above)
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/scripts/common.sh]: done

File: block-storage-attacher/tests/e2e/conf/prod-us-east.conf [hunk 9]
```diff
@@ -63,6 +65,7 @@ TEST_CLUSTER_CREATE=${TEST_CLUSTER_CREATE:-ifNotFound}    #Options: [ifNotFound
 TEST_CLUSTER_CONFIG_DOWNLOAD=true
 TEST_LATEST_IMAGE_BUILD=false
 TEST_HELM_INSTALL=true
+PORTWORX_HELM_INSTALL=true
```
Comment (akgunjal) [inline block-storage-attacher/tests/e2e/conf/prod-us-east.conf]: Remove this if not used

File: block-storage-attacher/tests/e2e/conf/stage.conf [hunk 10]
```diff
@@ -60,6 +61,7 @@ TEST_CLUSTER_CREATE=${TEST_CLUSTER_CREATE:-ifNotFound}    #Options: [ifNotFound
 TEST_CLUSTER_CONFIG_DOWNLOAD=true
 TEST_LATEST_IMAGE_BUILD=true
 TEST_HELM_INSTALL=true
+PORTWORX_HELM_INSTALL=true
```
Comment (akgunjal) [inline block-storage-attacher/tests/e2e/conf/stage.conf]: Remove this if not used

File: block-storage-attacher/tests/e2e/e2e-tests/e2eTests.txt [hunk 11]
```diff
@@ -0,0 +1,9 @@
+BlockVolumeAttacher-Volume-Test: Cluster-Creation: PASS
```
Comment (akgunjal) [inline block-storage-attacher/tests/e2e/e2e-tests/e2eTests.txt]: Why is this file added?

Comment (akgunjal) [reply block-storage-attacher/tests/e2e/e2e-tests/e2eTests.txt]: Also why is log file ` block-storage-attacher/tests/e2e/e2e-tests/log.txt` added?

File: block-storage-attacher/tests/e2e/e2e-tests/portworx_pvc.yaml (hunk 2 above)
Comment (akgunjal) [reply block-storage-attacher/tests/e2e/e2e-tests/portworx_pvc.yaml]: You said Done but the file is not removed from PR.

File: block-storage-attacher/tests/e2e/e2e-tests/portworx_values.yaml (hunk 4 above)
Comment (akgunjal) [reply block-storage-attacher/tests/e2e/e2e-tests/portworx_values.yaml]: no action taken on this comment. Please check and revert.

File: block-storage-attacher/tests/e2e/e2e.sh [hunk 12]
```diff
@@ -92,6 +92,11 @@ if [[ $TEST_HELM_INSTALL == "true" ]]; then
 	check_pod_state "ibm-block-storage-attacher" 
 	#check_daemonset_state "ibmcloud-block-storage-driver"
 fi
+#if [[ $PORTWORX_HELM_INSTALL == "true" ]]; then
```
Comment (akgunjal) [inline block-storage-attacher/tests/e2e/e2e.sh]: Remove this comments if not needed

File: block-storage-attacher/tests/e2e/scripts/common.sh (hunk 7 above)
Comment (akgunjal) [reply block-storage-attacher/tests/e2e/scripts/common.sh]: No action taken on this comment. Please check and revert.

File: block-storage-attacher/tests/e2e/scripts/common.sh (hunk 6 above)
Comment (akgunjal) [reply block-storage-attacher/tests/e2e/scripts/common.sh]: You said done but I see both the points given in comment are not taken care. So its not done.

File: block-storage-attacher/tests/e2e/e2e-tests/portworx_values.yaml (hunk 4 above)
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/e2e-tests/portworx_values.yaml]: Removed the file

File: block-storage-attacher/tests/e2e/scripts/common.sh (hunk 6 above)
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/scripts/common.sh]: Now doing with override values

File: block-storage-attacher/tests/e2e/scripts/common.sh (hunk 7 above)
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/scripts/common.sh]: We need not to provide the drives parameter as port worx has to identify the formatted drives

File: block-storage-attacher/tests/e2e/conf/stage.conf (hunk 10 above)
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/conf/stage.conf]: Removed

File: block-storage-attacher/tests/e2e/e2e-tests/e2eTests.txt (hunk 11 above)
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/e2e-tests/e2eTests.txt]: We will use this in future to provide the log file in case of failure

File: block-storage-attacher/tests/e2e/e2e.sh (hunk 12 above)
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/e2e.sh]: Removed

File: block-storage-attacher/tests/e2e/conf/prod-us-east.conf (hunk 9 above)
Comment (mumutyal) [reply block-storage-attacher/tests/e2e/conf/prod-us-east.conf]: Removed

File: block-storage-attacher/tests/e2e/e2e-tests/utilscript.sh [hunk 13]
```diff
@@ -23,8 +35,19 @@ elif [ "$2" = "voldelete" ]
...
+elif [ "$2" = "portworxpvcreate" ]
+then
+    install_portworx_plugin
+    check_portworx_pod_state "portworx"
+    echo "BlockVolumeAttacher-Volume-Test: PortWorx Plugin-Installation: PASS" >> $E2E_PATH/e2eTests.txt
+    export CLSFILE=$1
+    kubectl  create -f $CLSFILE
+elif [ "$2" = "portworxdelete" ]
+then
+    curl -fsL https://install.portworx.com/px-wipe | bash
```
Comment (akgunjal) [inline block-storage-attacher/tests/e2e/e2e-tests/utilscript.sh]: Just before curl command do `kubectl delete sc <storage class>`. Delete class so when u create in next run it wont error out.
